src/lib/Scrollama.js

```diff
@@ -24,7 +24,7 @@ const getPageHeight = () => {
 class Scrollama extends PureComponent {
   constructor(props) {
     super(props);
-    const { offset, debug, children, onStepEnter, onStepExit } = this.props;
+    const { offset, debug, children, onStepEnter, onStepExit, progressmode, onStepProgress, threshold } = this.props;
```

Comment:
`progressmode` should be in camelCase. If you don't want there to be a naming conflict, I suggest you name it something else.